packages/gestalt/src/TableRowExpandable.js

```diff
@@ -0,0 +1,51 @@
+// @flow strict
+import React, { type Node, useState } from 'react';
+import cx from 'classnames';
+import styles from './Table.css';
+import Box from './Box.js';
+import IconButton from './IconButton.js';
+
+type Props = {|
+  children: Node,
+  hoverStyle?: 'gray' | 'none',
+  expandedContents: Node,
+|};
+
+export default function TableRowExpandable(props: Props): Node {
+  const [expanded, setExpanded] = useState(false);
+  const [hovered, setHover] = useState(false);
+  const handleHover: (isHovered: boolean) => void = (isHovered: boolean) =>
+    setHover(isHovered);
+  const hoverStyle = props.hoverStyle || 'gray';
+  const cs =
+    hovered && hoverStyle === 'gray'
+      ? cx(styles.hoverShadeGray)
+      : cx(styles.hoverNone);
+
+  return (
+    <>
+      <tr
+        className={cs}
+        onMouseEnter={() => handleHover(true)}
+        onMouseLeave={() => handleHover(false)}
+      >
+        <td>
+          <IconButton
+            accessibilityLabel=""
```

Comment:
@faisalgedi we should pass these props on the root of the `Table.ExpandableRow` component:

```
accessibilityExpandLabel="Click to expand the row"
accessibilityCollapseLabel="Click to collapse the row"
```

both props should be required.

Reply:
`accessibilityLabel={expanded ? accessibilityCollapseLabel : accessibilityExpandLabel}`